### ISSUE TITLE
yaxt = 'n' should only suppress plotting of the y-axis; currently, it suppresses both

### ISSUE THREAD

**Issue by cjh57-cornell** [TomKellyGenetics/vioplot#16]
I am using R version 4.0.4 (2021-02-15) and vioplot_0.3.7.

`vioplot(outcome ~ intervention, data = d, xaxt = 'n')` works as expected: `intervention` values are not marked on the x-axis, but `outcome` values are marked on the y-axis. 

`vioplot(outcome ~ intervention, data = d, xaxt = 'n', horizontal = TRUE)` also does something understandable: `intervention` values are not marked on the (vertical) "x-axis", but outcome values are marked on the (horizontal) "y-axis." This is not what the current version of `boxplot` does, but it's possible to work around that.

But both `vioplot(outcome ~ intervention, data = d, yaxt = 'n')` and `vioplot(outcome ~ intervention, data = d, yaxt = 'n', horizontal = TRUE)` result in *neither* intervention values nor outcome values being marked on their respective axes. This seems like it has to be wrong.

I took a look at the code, and this clearly is at least partially a reflection of the fact that all the `if(xaxt !="n")` conditionals in vioplot.R are wrapped inside `if(yaxt !="n")` conditionals. However, when I added else clauses for the latter, with similar code for handling the x-axes alone inside it, this made the two snippets above work as I expected them to, but still resulted in this code:
```
par(yaxt = 'n')
vioplot(outcome ~ intervention, data = d)
```
suppressing the plotting of both axes. I do not know why this is.

**Comment by TomKellyGenetics** [comment]
Thanks for reporting this issue. The source code here is a little difficult to change since it is duplicated to handle log-scales. However, due to implementing the log-scales myself I am familiar with it.

I'm no longer actively developing this package but I will investigate this. A minor release with other changes is planned soon so it would be possible to incorporate this into the next release on CRAN.

As a workaround, it is possible to get the desired behavior by suppressing both axes and adding them as separate commands:

``` r
library("vioplot")
#> Loading required package: sm
#> Warning: package 'sm' was built under R version 4.1.3
#> Package 'sm', version 2.2-5.7: type help(sm) for summary information
#> Loading required package: zoo
#> Warning: package 'zoo' was built under R version 4.1.3
#> 
#> Attaching package: 'zoo'
#> The following objects are masked from 'package:base':
#> 
#>     as.Date, as.Date.numeric
outcome <- c(rnorm(25, 3, 1), rnorm(25, 2, 0.5))
intervention <- c(rep("treatment", 25), rep("control", 25))
table(intervention)
#> intervention
#>   control treatment 
#>        25        25
names(table(intervention))
#> [1] "control"   "treatment"
unique(sort(intervention))
#> [1] "control"   "treatment"
intervention <- as.factor(intervention)
levels(intervention)
#> [1] "control"   "treatment"
d <- data.frame(outcome, intervention)
vioplot(outcome ~ intervention, data = d, xaxt = 'n', yaxt = 'n')
axis(side = 1, at = 1:length(levels(intervention)), labels = levels(intervention))
```

![](https://i.imgur.com/s5qhA7N.png)

<sup>Created on 2022-11-09 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

I agree this is suboptimal, requiring tedious and cumbersome code. In fact it was one of the motivations to modify the source code to support log-scales directly which lead to development of vioplot v0.3.0 and taking over as maintainer.

**Comment by TomKellyGenetics** [comment]
Rather than adding more complexity to the if/else conditionals to add y-axes. It may be better to add x-axes separately. For example:

```r
if(yaxt =="n"){
    if(xaxt !="n"){
         Axis(1:length(datas), at = at, labels = label, side = 1, cex.axis = cex.names, col.axis = col.axis, font.axis = font.axis, mgp = mgp, xaxp = xaxp, tck = tck, tcl = tcl, las = las)
    }
}
```

Another solution is to refactor the yaxt conditionals so that only these lines are affected. This would allow parameters different for log-scales to be accounted for. Note the parameters names are the same for `horizontal = TRUE` but the plots are transposed.

```r
 if(yaxt !="n"){
    Axis(unlist(datas), side = 2, cex.axis = cex.axis, col.axis = col.axis, font.axis = font.axis, mgp = mgp, tck = tck, tcl = tcl, las = las) # xaxp = xaxp, yaxp = yaxp disabled for log
}

 if(yaxt !="n"){
    Axis(unlist(datas), side = 2, cex.axis = cex.axis, col.axis = col.axis, font.axis = font.axis, mgp = mgp, yaxp = yaxp, tck = tck, tcl = tcl, las = las)
}
```

**Comment by TomKellyGenetics** [comment]
I've tested this is corrected in the "dev" branch. It will be fixed in the next release (0.4.0) on CRAN.
